serverless-logic/serverless-logic-supporting-services/serverless-logic-data-index.adoc

```diff
@@ -8,4 +8,19 @@ toc::[]
 
 The Data Index service is a dedicated supporting service that stores the data related to the workflow instances and their associated jobs. This service provides a GraphQL endpoint allowing users to query and modify that data.
 
-//I will include more modules in smaller, incremental PRs to avoid having one large PR. Currently building the directory structure for the OSL content.
\ No newline at end of file
+The Data Index service processes data received through events, which can originate from any workflow or directly from the Job service. You can configure this event communication in various ways, as outlined in the Data Index communication configuration section.
+
+Data Index supports Apache Kafka or Knative Eventing to consume CloudEvents messages from workflows. It indexes and stores this event data in a database, making it accessible through GraphQL. These events provide detailed information about the units of work executed by a workflow. The Data Index service is central to {ServerlessLogicProductName} search, insights, and management capabilities.
+
+The key features of the Data Index service are as follows:
+
+* A flexible data structure
+* A distributable, cloud-ready format
+* Message based communication with workflows via Apache Kafka, Knative, and CloudEvents
+* A powerful GraphQL based querying API
+* Management capabilities through the Gateway API, enabling GraphQL actions on remote runtimes via a single entry point
+
```

Comment:
```suggestion
[NOTE]
====
When you are using the {ServerlessOperatorName} to deploy workflows, there is no need to manually install or configure the Data Index service. The Operator handles these tasks automatically and manages all necessary configurations for each workflow to connect with it.
====
```

Reply:
@kaldesai  I think that in this note, we should add a link to the upcoming supporting services section.
So, we have so we have something like this:

.... all necessary configurations for each workflow to connect with it, **see**" 👈 and here we add the link